package_modules/yum

```diff
@@ -0,0 +1,190 @@
+    for line in sys.stdin:
+        if line.startswith("File="):
+            pkg_string = line.split("=", 1)[1].rstrip()
+            break
+
+    if not pkg_string:
+        return 1
+
+    repo = False
+    if pkg_string.startswith("/"):
+        # Absolute file.
+        sys.stdout.write("PackageType=file\n")
+        sys.stdout.flush()
+        return subprocess.call([rpm_cmd, "--qf", rpm_output_format, "-qp", result.group(1)])
+    elif re.search("([:,]|-[0-9])", pkg_string):
+        # Contains either a version number or an illegal symbol.
+        sys.stdout.write(line + "ErrorMessage: Package string with illegal format\n")
+        return 1
+    else:
+        repo = True
+        sys.stdout.write("PackageType=repo\n")
+        sys.stdout.write("Name=" + pkg_string + "\n")
+        return 0
+
+
+def list_installed():
+    # Ignore everything.
+    sys.stdin.readlines()
+
+    return subprocess.call([rpm_cmd, "-qa", "--qf", rpm_output_format])
+
+
+def list_updates(online):
+    # Ignore everything.
+    sys.stdin.readlines()
+
+    online_flag = [] if online else ["-C"]
+
+    process = subprocess.Popen([yum_cmd, yum_options] + online_flag + ["check-update"], stdout=subprocess.PIPE)
+    lastline = ""
+    for line in process.stdout:
+        # Combine multiline entries into one line. A line without at least three
+        # space separated fields gets combined with the next line, if that line
+        # starts with a space.
+        if lastline and not line[0].isspace():
+            # Line does not start with a space. No combination.
+            lastline = ""
+
+        line = lastline + line
+        match = re.match("^\S+\s+\S+\s+\S+", line)
+        if match == None:
+            # Keep line but strip trailing newline.
+            lastline = line[:-1]
+            continue
+
+        lastline = ""
+        match = re.match("^(?P<name>\S+)\.(?P<arch>[^.\s]+)\s+(?P<version>\S+)\s+\S+\s*$", line)
+        if match != None:
+            sys.stdout.write("Name=" + match.group("name") + "\n")
+            sys.stdout.write("Version=" + match.group("version") + "\n")
+            sys.stdout.write("Architecture=" + match.group("arch") + "\n")
+
+    return 0
+
+
+def package_arguments_builder():
+    name = ""
+    version = ""
+    arch = ""
+    args = []
+    for line in sys.stdin:
+        if line.startswith("Name="):
+            if name:
+                # Each new "Name=" triggers a new entry.
+                args_add = name
+                if version:
+                    args_add += "-" + version
+                if arch:
+                    args_add += ":" + arch
+                args.append(args_add)
+
+                version = ""
+                arch = ""
+
+            name = line.split("=", 1)[1].rstrip()
+
+        elif line.startswith("Version="):
+            version = line.split("=", 1)[1].rstrip()
+
+        elif line.startswith("Architecture="):
+            arch = line.split("=", 1)[1].rstrip()
+
+    if name:
```

Comment:
Following my RedHat fixes (which I shall push soon), this section is completely rearranged.